atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java

```diff
@@ -101,11 +100,33 @@ private InputStream tryGetStream(Transaction transaction, ID id, StreamMetadata
             loadSingleBlockToOutputStream(transaction, id, 0, ios);
             return ios.getInputStream();
         } else {
-            File file = loadToNewTempFile(transaction, id, metadata);
-            return new DeleteOnCloseFileInputStream(file);
+            return makeStream(id, metadata);
```

Comment:
The first stream block loaded is from a new transaction, whereas this method already receives an open transaction. Is this intended?